Source/SpiderEye.Core/Bridge/IWebviewBridge.cs

```diff
@@ -9,18 +8,37 @@ namespace SpiderEye.Bridge
     /// </summary>
     public interface IWebviewBridge
     {
+        /// <summary>
+        /// Gets a value indicating whether dependency injection is enabled for the bridge.
+        /// If this is true, bridge calls run in a scoped DI lifetime.
+        /// </summary>
+        public bool IsDependencyInjectionEnabled { get; }
+
         /// <summary>
         /// Adds a custom handler to be called from the webview.
         /// </summary>
         /// <param name="handler">The handler instance.</param>
         void AddHandler(object handler);
 
+        /// <summary>
+        /// Adds a custom handler to be called from the webview.
+        /// Note: This method throws if <see cref="IsDependencyInjectionEnabled"/> is false.
+        /// </summary>
+        /// <typeparam name="T">The handler type.</typeparam>
+        void AddHandler<T>();
+
         /// <summary>
         /// Adds or replaces a custom handler to be called from the webview.
         /// </summary>
         /// <param name="handler">The handler instance.</param>
         void AddOrReplaceHandler(object handler);
 
+        /// <summary>
+        /// Adds or replaces a custom handler to be called from the webview.
+        /// </summary>
+        /// <typeparam name="T">The handler type.</typeparam>
+        void AddOrReplaceHandler<T>();
```

Comment:
Is this still needed with DI?

Reply:
If you want to register services only for one window (not for all), then this would still be needed.